source/extensions/quic_listeners/quiche/envoy_quic_client_stream.cc

```diff
@@ -158,18 +162,17 @@ void EnvoyQuicClientStream::OnBodyAvailable() {
     buffer->commit(&slice, 1);
     MarkConsumed(bytes_read);
   }
+  ASSERT(buffer->length() == 0 || !end_stream_decoded_);
 
-  // True if no trailer and FIN read.
-  bool finished_reading = IsDoneReading();
-  bool empty_payload_with_fin = buffer->length() == 0 && fin_received();
+  bool fin_read_and_no_trailers = IsDoneReading();
```

Comment:
So if we receive trailers first, then all the data, IsDoneReading will return false even though we have read all the data for this stream?  Should that be renamed?

Reply:
Nope. IsDoneReading() returning false in this case makes sense, because trailers has not been consumed yet. trailers should only be consumed after all the data is consumed.